experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+                            break  # we have changed the 'clrfrqrange' - must restart the for loop
+                            # checking in case it's in another range.
+                        else:  # neither end of clrfrqrange is inside the restricted range but
+                            # we should check if the range is inside the clrfrqrange.
+                            try:
+                                assert freq_range[0] not in range(exp_slice['clrfrqrange'][0],
+                                                                  exp_slice['clrfrqrange'][1])
+                            except AssertionError:
+                                # TODO Log a warning that there is a restricted range in the middle
+                                # of the
+                                # clrfrqrange that will be avoided OR could make this an Error.
+                                # Still need to implement clear frequency searching.
+                                pass
+                else:  # no break, so no changes to the clrfrqrange
+                    still_checking = False
+
+        elif exp_slice['rxonly']:  # RX only mode.
+            # In this mode, rxfreq is required.
+            try:
+                assert isinstance(exp_slice['rxfreq'], int) or isinstance(exp_slice['rxfreq'],
+                                                                          float)
+                assert (exp_slice['rxfreq'] * 1000) < self.options.max_freq
+                assert (exp_slice['rxfreq'] * 1000) > self.options.min_freq
+            except AssertionError:
+                errmsg = """rxfreq must be a number (kHz) between min and max frequencies for the 
+                            radar"""
+                raise ExperimentException(errmsg)
+
+        else:  # TX-specific mode , without a clear frequency search.
+            # In this mode, txfreq is required along with the other requirements.
+            try:
+                assert isinstance(exp_slice['txfreq'], int) or isinstance(exp_slice['txfreq'],
+                                                                          float)
+                assert (exp_slice['txfreq'] * 1000) < self.options.max_freq
+                assert (exp_slice['txfreq'] * 1000) > self.options.min_freq
+            except AssertionError:
+                errmsg = """txfreq must be a number (kHz) between min and max frequencies for the 
+                                            radar"""
+                raise ExperimentException(errmsg)
+            for freq_range in self.options.restricted_ranges:
+                try:
+                    assert exp_slice['txfreq'] not in range(freq_range[0], freq_range[1])
+                except AssertionError:
+                    errmsg = """txfreq is within a restricted frequency range"""
+                    raise ExperimentException(errmsg)
+                    # TODO or could just log this and set to default frequency, or the closest
```

Comment:
Ok - we could potentially fail and go to default program (common mode) - that could be handled in the scheduler.